typescript/web/src/components/import-button/import-images-modal/types.ts

```diff
@@ -22,3 +22,13 @@ export type DroppedUrl = {
  * the file being uploaded
  */
 export type UploadStatuses = Record<string, boolean | string>;
+
+/**
+ * Generic type of an useState setter function
+ */
+type SetState<T> = (newStateOrUpdater: T | ((oldState: T) => T)) => void;
```

Comment:
Isn't there a standard type already covering this case?

Reply:
React already provides but it's something like `React.Dispatch<React.SetStateAction<T>>`. I felt like it's too tied to `useStates`, but I can use this one if you prefer.

Reply:
Yes, it is equivalent to `Dispatch<SetStateAction>`:

```ts
    // Unlike the class component setState, the updates are not allowed to be partial
    type SetStateAction<S> = S | ((prevState: S) => S);
    // this technically does accept a second argument, but it's already under a deprecation warning
    // and it's not even released so probably better to not define it.
    type Dispatch<A> = (value: A) => void;
```

It's still preferable to reuse the existing types instead of redefining them. Even more in such a specific file.

 Given that there is no shorthand for `Dispatch<SetStateAction>`, you could still add (in e.g. `typescript/web/src/utils/types/set-state.ts` - with a proper `index.ts` chain so that we don't have to use the full path to import it 😉 ) the following type definition if you believe it'll be too much repeated in the code:

 ```ts
 import { Dispatch, SetStateAction } from "react";
 
 type SetState<TValue> = Dispatch<SetStateAction<TValue>>;
 ```